indexer.ts

```diff
@@ -0,0 +1,743 @@
+    console.log("new Registry data::", { ...data }, receiverId);
+
+    if (data.admins) {
+      for (const admin in data.admins) {
+        await context.db.Account.upsert({ id: admin }, ["id"], []);
+        let list_admin = {
+          list_id: receiverId,
+          admin_id: admin,
+        };
+        await context.db.ListAdmin.insert(list_admin);
+      }
+    }
+
+    let regv = {
+      id: receiverId,
+      owner_id: data.owner,
+      default_registration_status: "Approved", // the first registry contract has approved as default, and the later changed through the admin set function call, which we also listen to, so it should self correct.
+      name: receiverId.split(".")[0],
+      tx_hash: receiptId,
+    };
+
+
+    await context.db.Account.upsert({ id: data.owner }, ["id"], []);
+
+    await context.db.List.insert(regv);
+  }
+
+  // function tracks register function calls on the registry conract
+  async function handleNewProject(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("new Project data::", { ...data }, receiverId);
+    let registry = (await context.db.List.select({ id: receiverId }))[0]; // fetch the registry so as to get default
+    let reg = {
+      registrant_id: data._project_id || signerId,
+      status: registry.default_registration_status,
+      submitted_at: new Date(
+        Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+      ),
+    };
+    if (data._project_id) {
+      await context.db.Account.upsert({ id: data._project_id }, ["id"], []);
+    }
+
+    await context.db.Account.upsert({ id: signerId }, ["id"], []);
+
+    await context.db.ListRegistration.insert(reg);
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: reg.submitted_at,
+      type: "Register",
+      action_result: reg.registrant_id,
+      tx_hash: receiptId,
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  async function handleProjectRegistrationUpdate(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new Project data::", { ...data }, receiverId);
+    let regUpdate = {
+      status: data.status,
+      admin_notes: data.review_notes,
+      updated_at: new Date(
+        Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+      ),
+    };
+
+    await context.db.ListRegistration.update(
+      { registrant_id: data.project_id },
+      regUpdate
+    );
+  }
+
+  async function handlePotApplication(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("new factory data::", { ...data }, receiverId);
+    await context.db.Account.upsert({ id: data.project_id }, ["id"], []);
+    let application = {
+      pot_id: receiverId,
+      applicant_id: data.project_id,
+      message: data.message,
+      submitted_at: data.submitted_at,
+      current_status: data.status,
```

Comment:
this should be taken from the result of the method call (the new application) as the `status` won't be present in the args, and can vary depending on the default status that is currently set